KeyboardSupport/Classes/KeyboardSupportViewController.swift

```diff
@@ -0,0 +1,190 @@
+//
+//  KeyboardSupportViewController.swift
+//  KeyboardSupport
+//
+//  Copyright © 2018 Bottle Rocket Studios. All rights reserved.
+//
+
+import UIKit
+
+/// Callback when the "Done" button is tapped.
+public protocol KeyboardSupportDelegate: class {
+    func didTapDoneButton()
+}
+
+/// A subclass of UIViewController for making keyboard interactions easier.
+open class KeyboardSupportViewController: UIViewController {
+    
+    // MARK: - Properties
+    
+    private var textFields: [UITextField] = []
+    private var scrollView: UIScrollView?
+    private var bottomConstraint: NSLayoutConstraint?
+    private var constraintOffset: CGFloat = 0
+    private var usesDismissalView: Bool = false
+    private var usesKeyboardNextButtons: Bool = false
+    private var keyboardInputAccessoryView: KeyboardInputAccessoryView?
+    private var currentTextField: UITextField?
+    public weak var keyboardSupportDelegate: KeyboardSupportDelegate?
+    
+    // MARK: - Lifecycle
+    
+    open override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        NotificationCenter.default.addObserver(self, selector: #selector(keyboardWillShow), name: .UIKeyboardWillShow, object: nil)
+        NotificationCenter.default.addObserver(self, selector: #selector(keyboardWillHide), name: .UIKeyboardWillHide, object: nil)
+    }
+    
+    deinit {
+        NotificationCenter.default.removeObserver(self, name: .UIKeyboardWillShow, object: nil)
+        NotificationCenter.default.removeObserver(self, name: .UIKeyboardWillHide, object: nil)
+    }
+    
+    // MARK: - Public
+    
+    /// Configures different types of support for the keyboard.
+    ///
+    /// - Parameters:
+    ///   - textFields: UITextfields whose order matters for navigating from first to last.
+    ///   - scrollView: A UIScrollView whose contentInset is updated so the current text field is not hidden. Optional.
+    ///   - bottomConstraint: An NSLayoutConstraint to move a view above the keyboard. Optional.
+    ///   - constraintOffset: A value to offset the bottomConstraint.
+    ///   - usesDismissalView: Allows the keyboard to be dismissed by tapping outside a text field. Default is false.
+    ///   - usesKeyboardNextButtons: Allows the keyboard's "return" key to be used to navigate between text fields. Default is false.
+    ///   - keyboardInputAccessoryView: A KeyboardInputAccessoryView to show above the keyboard to navigate between text fields. Optional.
+    final public func configureKeyboardSupport(with textFields: [UITextField], scrollView: UIScrollView?, bottomConstraint: NSLayoutConstraint?, constraintOffset: CGFloat = 0, usesDismissalView: Bool, usesKeyboardNextButtons: Bool, keyboardInputAccessoryView: KeyboardInputAccessoryView? = nil) {
```

Comment:
There is a ton of arguments to this function - can we break it up at all? Either with optional parameters and/or some kind of KeyboardConfiguration struct?

Reply:
I'll add a struct and an init with optional parameters